plugins/woocommerce-blocks/docs/third-party-developers/extensibility/register-product-collection.md

```diff
@@ -0,0 +1,235 @@
+# Register Product Collection
+
+`__experimentalRegisterProductCollection` function is part of the `@woocommerce/blocks-registry` package. This function allows 3PDs to register a new collection. This function accepts most of the arguments that are accepted by [Block Variation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/#defining-a-block-variation).
+
+> [!WARNING]
+> It's experimental and may change in the future. Please use it with caution.
+
+**There are two ways to use this function:**
+
+1. Using `@woocommerce/dependency-extraction-webpack-plugin` in a Webpack configuration: This will allow you to import the function from the package & use it in your code. For example:
+
+	```tsx
+	import { __experimentalRegisterProductCollection } from "@woocommerce/blocks-registry";
+	```
+
+2. Using the global `wc` object: This will allow you to use the function using the global JS object without importing it. For example:
+
+	```tsx
+	wc.wcBlocksRegistry.__experimentalRegisterProductCollection({
+	...
+	});
+	```
+
+> [!TIP]
+> The first method is recommended if you are using Webpack.
+
+## Examples of how to use `__experimentalRegisterProductCollection`
+
+### Example 1: Registering a new collection
+
+```tsx
+__experimentalRegisterProductCollection({
+  name: "woocommerce/product-collection/my-custom-collection",
```

Comment:
What about using the plugin-name here? And similar to the others.

```suggestion
  name: "plugin-name/product-collection/my-custom-collection",
```

Reply:
Good idea! As 3PDs will copy paste these examples, it's better to use `plugin-name` as placeholder. I'll update it 🚀